pygmt/tests/test_legend.py

```diff
@@ -44,42 +43,25 @@ def test_legend_default_position():
     return fig
 
 
-@check_figures_equal()
+@pytest.mark.mpl_image_compare
 def test_legend_entries():
```

Comment:
Rewrite the tests using a simpler 1d array, rather than the `@Table_5_11.txt` file?

Reply:
I agree, but best to not modify the test too much in this PR as mentioned in https://github.com/GenericMappingTools/pygmt/pull/1104#discussion_r602082727